### PR TITLE
Support for "forward" statement

### PR DESCRIPTION
**Pull request by Hossy**
Add "forward" statement in options { } block in named.conf.

### CODE REVIEW

File: templates/named.conf.erb [hunk 1]
```diff
@@ -21,6 +21,9 @@ options {
 <%-   end -%>
 	};
 <%- end -%>
+<%- if @forward != '' -%>
```
Comment (inkblot) [inline templates/named.conf.erb]: Does `undef == ''`? I'm trying to work out what happens when this parameter is not set and retains its default value of `undef`, but puppet's types have always been unlike anything else.

Comment (Hossy) [inline templates/named.conf.erb]: You have a good point.  I think I originally just copied the line from zone.conf.erb, but looking more into it, you've used "<%- if @variable and @variable != '' -%>" as well which may be a better approach.

Also, this being the first time I've worked with a puppet module, will the reuse of @forward (from zone.conf.erb) create a conflict or is it unique to the scope of the conf file?

Comment (Hossy) [inline templates/named.conf.erb]: I found this which suggests that @variable != '' isn't even needed, but would definitely need testing.  http://grokbase.com/t/gg/puppet-users/147by1key3/checking-a-variable-is-not-undef

Comment (Hossy) [inline templates/named.conf.erb]: To be clear here, "<%- if @variable and @variable != '' -%>" will be necessary to handle the possibility of undef per the [Puppet 4.1+ documentation](https://docs.puppet.com/puppet/4.8/reference/lang_data_boolean.html):

"If a non-boolean value is used where a boolean is required:
* The undef value is converted to boolean false.
* All other values are converted to boolean true.

Notably, this means the string values "" (zero-length string) and "false" both resolve to true."

**Comment by inkblot** [comment]
Thanks! I may rework this a little bit after merging, but it looks like working code.

**Comment by inkblot** [comment]
This change is included in version 7.1.0 of the module on the puppet forge.

**Pull request merged**